IdentityServer/v7/docs/content/reference/services/token_creation_service.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Token Creation Service"
+weight: 50
+---
+
+IdentityServer uses an *ITokenCreationService* which is responsible for the creation
+of tokens, with the default implementation of *DefaultTokenCreationService*.
+
+```csharp
+/// <summary>
+/// Logic for creating security tokens
+/// </summary>
+public interface ITokenCreationService
+{
+    /// <summary>
+    /// Creates a token.
+    /// </summary>
+    /// <param name="token">The token description.</param>
+    /// <returns>A protected and serialized security token</returns>
+    Task<string> CreateTokenAsync(Token token);
+}
+```
+
+The Token creation service involves taking the *Token* model and converting it into
+a JWT. During the JWT creation, you have one last opportunity to
+modify the *Token* by adding, removing, or altering property values. Everyday use cases
+for implementing the *ITokenCreationService* include modifying claims, audiences, and more
```

Comment:
I would stress that the token creation service is best used in advanced cases where you need to customize the serialization of a token or in the rare case that the `Token` model doesn't support what you need.

- ITokenCreateService: Serialization of the model
- ITokenService: Building the model of a token (more than claims)
- IClaimsService: Customizing claims for tokens
- IProfileService: User-centric profile data (gets used in tokens and also the userinfo endpoint)